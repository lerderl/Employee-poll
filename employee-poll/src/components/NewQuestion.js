import { connect } from "react-redux";
import { useEffect, useState } from "react";

import Authenticate from "./Authenticate";
import { handleAddQuestion } from "../actions/questions";

const NewQuestion = ({ dispatch, authedUser }) => {
  const [firstOption, setFirstOption] = useState('');
  const [secondOption, setSecondOption] = useState('');
  const [success, setSuccess] = useState(false);
  const [error, setError] = useState(false);

  useEffect(() => {
    console.log({ success, error });
  }, [success, error]);

  const handleSubmit = e => {
    e.preventDefault();

    if (firstOption !== '' & secondOption !== '') {
      const question = {
        optionOneText: firstOption,
        optionTwoText: secondOption
      };

      dispatch(handleAddQuestion(question));
      setSuccess(true);
      setError(false);
    } else {
      setSuccess(false);
      setError(true);
    };

    setFirstOption('');
    setSecondOption('');
  };

  const optionOneChange = e => {
    setFirstOption(e.target.value);
  };

  const optionTwoChange = e => {
    setSecondOption(e.target.value);
  };

  return (
    <div className="container text-center">
      <Authenticate />
      {success && (
        <h1 data-testid="success-header">Question saved successfully</h1>
      )}
      {error && (
        <h1 data-testid="error-header">Error: Something went wrong.</h1>
      )}
      <h1 className="text-dark">Would you rather</h1>
      <h6 className="text-secondary">Create your own poll</h6>
      <form onSubmit={handleSubmit}>
        <div class="mb-3">
          <label htmlFor="firstOptionValue" class="form-label">
            First Option
          </label>
          <input
            value={firstOption}
            class="form-control"
            name="firstOptionValue"
            placeholder="Option One"
            onChange={optionOneChange}
            data-testid="option-one-input"
            required
          />
        </div>
        <div class="mb-3">
          <label htmlFor="secondOptionValue" class="form-label">
            Second option
          </label>
          <input
            value={secondOption}
            class="form-control"
            name="secondOptionValue"
            placeholder="Option Two"
            onChange={optionTwoChange}
            data-testid="option-two-input"
            required
          />
        </div>
        <button
          type="submit"
          className="btn btn-primary"
          data-testid="submit-button"
        >
          Submit
        </button>
      </form>
    </div>
  );
};

const mapStateToProps = ({ authedUser }) => {
  return {
    authedUser
  };
};

export default connect(mapStateToProps)(NewQuestion);
