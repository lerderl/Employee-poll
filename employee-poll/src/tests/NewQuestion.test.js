import { Provider } from "react-redux";
import { legacy_createStore } from "redux";
import { MemoryRouter } from "react-router-dom";
import { render, screen, fireEvent } from "@testing-library/react";

import reducer from "../reducers";
import middleware from "../middleware";
import NewQuestion from "../components/NewQuestion";

const store = legacy_createStore(reducer, middleware);

describe("NewQuestion", () => {
  it("will save new question.", () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <NewQuestion />
        </Provider>
      </MemoryRouter>
    );

    var optionOne = screen.getByTestId("option-one-input");
    fireEvent.change(optionOne, { target: { value: "Do you know this?" } });
    var optionTwo = screen.getByTestId("option-two-input");
    fireEvent.change(optionTwo, { target: { value: "Or this?" } });
    var submitButton = screen.getByTestId("submit-button");
    fireEvent.click(submitButton);
    expect(screen.getByTestId("success-header")).toBeInTheDocument();
    expect(screen.queryByTestId("error-header")).not.toBeInTheDocument();
  });

  it("should match snapshot", () => {
    var { component } = render(
      <MemoryRouter>
        <Provider store={store}>
          <NewQuestion />
        </Provider>
      </MemoryRouter>
    );

    expect(component).toMatchSnapshot();
  });
});
