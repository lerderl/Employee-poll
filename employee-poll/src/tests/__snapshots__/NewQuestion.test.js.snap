// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewQuestion should match snapshot 1`] = `undefined`;
